source/val/validate_decorations.cpp

```diff
@@ -848,9 +841,40 @@ spv_result_t CheckDecorationsOfEntryPoints(ValidationState_t& vstate) {
             break;
           if (auto error = CheckBuiltInVariable(interface, vstate))
             return error;
-        } else if (isBuiltInVar(interface, vstate)) {
-          if (auto error = CheckBuiltInVariable(interface, vstate))
-            return error;
+        } else {
```

Comment:
This misses the case that you have duplicates between the built-in block and loose variables. I expect the built-in block part of this if should also record which builtins are used appropriately.

Reply:
I thought the [whole idea](https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/6625#note_467693) of built-in block is the `builtin` is tied to the `OpTypeStruct` and not the `OpVariable`  so we don't need to check those

Reply:
The built-in block is still part of the interface and the point of that change is to say no builtin duplication in the interface. For a given interface, all the builtins may be on variables, but some might also be in a builtin block.

Reply:
re-worked, section of code, added tests as well